nengo_gui/exec_env.py

```diff
@@ -117,6 +117,11 @@ def __exit__(self, exc_type, exc_value, traceback):
 
         sys.stdout = sys.__stdout__
 
+        # ensure what has been printed is safe to show in html
+        s = self.stdout.getvalue()
+        s = s.replace('<', '&lt;').replace('>', '&gt;')
```

Comment:
This happens before the link is added.  This is to make sure that anything that gets printed with normal print statements doesn't get turned into an HTML tag.  This used to happen automatically because we used `.text` on the javascript side, but we can't do that now because that would change *everything* and make the link not work.

So the process here is to let all the print statements happen, convert them to be HTML-safe, and then after that `exec_env.stdout.write` in `page.py` will happen.  Very confusing.  :(